### PR TITLE
feat(frontend): add quiz question component bb-88

### PR DESCRIPTION
**Pull request by yulyaolshanska**
### Summary
This PR implements the quiz question component.

### Created
- quiz question component

### Changes
- **input.tsx:** Added type radio
- **input styles** Added styles for radio input

image

### CODE REVIEW

File: apps/frontend/src/libs/components/input/input.tsx [hunk 1]
```diff
@@ -6,42 +6,75 @@ import {
...
+				{label}
+			</span>
 			<div className={styles["input-container"]}>
-				<input
-					className={getValidClassNames(styles["input-field"])}
-					{...field}
-					placeholder={placeholder}
-					type={type}
-				/>
+				{type === "radio" && options ? (
```
Comment (revenets) [inline apps/frontend/src/libs/components/input/input.tsx:55]: And move it to a separate variable please
```suggestion
				{type === "radio" && options.length ? (
```

File: apps/frontend/src/libs/components/input/input.tsx [hunk 2]
```diff
@@ -6,42 +6,75 @@ import {
...
 	label,
 	name,
+	options,
 	placeholder = "",
 	type = "text",
 }: Properties<T>): JSX.Element => {
 	const { field } = useFormController({ control, name });
 
-	const error = errors[name]?.message;
+	const error = errors && errors[name]?.message;
```
Comment (revenets) [inline apps/frontend/src/libs/components/input/input.tsx:34]:
```suggestion
	const error = errors?.[name]?.message;
```

File: apps/frontend/src/libs/components/app/app.tsx [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
 } from "~/libs/hooks/hooks.js";
 import { actions as userActions } from "~/modules/users/users.js";
 
+type FormValues = {
+	value: string;
+};
+
 const App: React.FC = () => {
 	const { pathname } = useLocation();
+	const { control } = useAppForm<FormValues>({
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/app/app.tsx:23]: Create a separate quiz component and handle all logic there

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/app/app.tsx (hunk 3 above)
Comment (yulyaolshanska) [reply apps/frontend/src/libs/components/app/app.tsx:23]: Should quiz question component be in folder with new quiz component(quiz component -> components -> quiz question component) or in separate folders?

Comment (fshabanov) [reply apps/frontend/src/libs/components/app/app.tsx:23]: Should be in ~/pages/quiz/libs/components

Comment (yulyaolshanska) [reply apps/frontend/src/libs/components/app/app.tsx:23]: Should I add mock data?

Comment (fshabanov) [reply apps/frontend/src/libs/components/app/app.tsx:23]: We already have PR for adding questions, consult with that developer to get a real type. You can mock data for now, but use a real type

File: apps/frontend/src/libs/components/app/app.tsx [hunk 4]
```diff
@@ -12,6 +12,8 @@ import {
 } from "~/libs/hooks/hooks.js";
 import { actions as userActions } from "~/modules/users/users.js";
 
+import { QuizForm } from "../../../pages/quiz/libs/components/quiz-form/quiz-form.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/app/app.tsx:15]:
```suggestion
import { QuizForm } from "~/pages/quiz/libs/components/quiz-form/quiz-form.js";
```

File: apps/frontend/src/libs/types/form.types.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export {
```
Comment (fshabanov) [inline apps/frontend/src/libs/types/form.types.ts:1]: No need to have this file, export directly in types.ts file

File: apps/frontend/src/libs/components/input/input.tsx [hunk 6]
```diff
@@ -6,42 +6,76 @@ import {
 } from "react-hook-form";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/input/input.tsx:6]: Change imports here as well, since we now export those from ~/libs/types/types.ts

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/app/app.tsx (hunk 4 above)
Comment (yulyaolshanska) [reply apps/frontend/src/libs/components/app/app.tsx:15]: I tryed this but got error 
image

Comment (fshabanov) [reply apps/frontend/src/libs/components/app/app.tsx:15]: The path seems valid to me, try to find out what is the issue

**Review by WilhelmWesser — APPROVED** [review]

File: apps/frontend/src/pages/quiz/libs/components/quiz-form/quiz-form.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { QuizQuestion } from "~/libs/components/components.js";
+import { useAppForm } from "~/libs/hooks/hooks.js";
+
+type FormValues = {
+	value: string;
+};
```
Comment (fshabanov) [inline apps/frontend/src/pages/quiz/libs/components/quiz-form/quiz-form.tsx:6]: This seems like incorrect type, we will have multiple questions, so we should be able to save multiple answers

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/quiz/libs/components/quiz-form/quiz-form.tsx (hunk 7 above)
Comment (yulyaolshanska) [reply apps/frontend/src/pages/quiz/libs/components/quiz-form/quiz-form.tsx:6]: I thought the form would be in another task, so I added this just to add my quiz question component.

Comment (fshabanov) [reply apps/frontend/src/pages/quiz/libs/components/quiz-form/quiz-form.tsx:6]: Understood, let's keep in that case

**Review by fshabanov — APPROVED** [review]

**Pull request merged**